File: app/src/main/java/com/club360fit/app/data/SupabaseClient.kt
package com.club360fit.app.data

import com.club360fit.app.BuildConfig
import io.github.jan.supabase.createSupabaseClient
import io.github.jan.supabase.functions.Functions
import io.github.jan.supabase.gotrue.Auth
import io.github.jan.supabase.postgrest.Postgrest
import io.github.jan.supabase.storage.Storage

/**
 * Supabase client singleton. URL comes from BuildConfig (default matches iOS + dashboard; override via local.properties SUPABASE_URL).
 * Anon key is read from BuildConfig (set via local.properties).
 * For profile avatars, create a public bucket named "avatars" in Supabase Storage.
 */
object SupabaseClient {

    const val AVATARS_BUCKET = "avatars"
    const val TRANSFORMATIONS_BUCKET = "transformations"
    /** Daily meal photos (client uploads; coach reviews). Create in Supabase Storage. */
    const val MEAL_PHOTOS_BUCKET = "meal-photos"

    val client = createSupabaseClient(
        supabaseUrl = BuildConfig.SUPABASE_URL,
        supabaseKey = BuildConfig.SUPABASE_ANON_KEY
    ) {
        install(Auth) {
            // Used for password reset deep links: club360fit://reset
            scheme = "club360fit"
            host = "reset"
        }
        install(Postgrest)
        install(Functions)
        install(Storage)
    }
}

File: Club360Fit-iOS/Club360fit/Club360fit/SupabaseClient.swift
import Foundation
import Supabase

/// Shared Supabase client — mirrors Android `SupabaseClient.kt` (same project URL & anon key).
enum Club360FitSupabase: Sendable {
    /// Same bucket id as Android `SupabaseClient.MEAL_PHOTOS_BUCKET`.
    static let mealPhotosBucket = "meal-photos"
    /// Android `SupabaseClient.TRANSFORMATIONS_BUCKET`.
    static let transformationsBucket = "transformations"
    /// Android `SupabaseClient.AVATARS_BUCKET`.
    static let avatarsBucket = "avatars"

    static let shared = SupabaseClient(
        supabaseURL: AppConfig.supabaseURL,
        supabaseKey: AppConfig.supabaseAnonKey
    )

    /// Password recovery / deep links (`club360fit://reset?...`). Keeps `import Supabase` out of `Club360fitApp.swift`.
    static func handleAuthRedirectURL(_ url: URL) {
        shared.auth.handle(url)
    }

    /// Options for Edge Functions that call `getUser()` with the caller’s JWT (`set-user-role`, `transfer-client`, etc.).
    /// Ensures `Authorization` is the signed-in user’s access token, not only the anon key.
    static func functionInvokeOptions<T: Encodable>(body: T) async throws -> FunctionInvokeOptions {
        let session = try await shared.auth.session
        return FunctionInvokeOptions(
            headers: ["Authorization": "Bearer \(session.accessToken)"],
            body: body
        )
    }
}
